### ISSUE TITLE
[Staging] Regional Page - and other pages - stuck on loading

### ISSUE THREAD

**Issue by nanometrenat** [IFRCGo/go-frontend#1491]
## Issue
Regional pages on GO are stuck on loading - much of the page loads but then there are the 3 loading dots still going forever.

@mariam-yu raised this first, having tried in an incognito window also - still same issue.  Screenshot below, please investigate.  I will put har file in Teams FYI - the only weird things I can spot in Network tab is that https://dsgocdnapi.azureedge.net/api/v2/featured_event_deployments is returning 301 error and https://dsgoproxyapp.azurewebsites.net/regions/1 is returning 304.  Not sure if either of these explain the issue.

## URL
https://dsgoproxyapp.azurewebsites.net/regions/1#operations

## Steps to reproduce
![image](https://user-images.githubusercontent.com/33625192/89309673-0106e180-d674-11ea-81a7-3ead0c3e484c.png)

## Expected behaviour
Page should load!

## Related feature
Unknown

## Criticality/Urgency
Critical but staging only.

**Comment by batpad** [comment]
Thanks for this catch!

I can reproduce this sometimes on the develop branch as well. @frozenhelium my hunch is it might have to do with the init changes, but I'm not certain. I don't see any obvious errors in the console. I'll dig into this further tomorrow. @frozenhelium, am co-assigning you, but feel free to unassign yourself if you're certain this is not due to the init changes. I need to dig further to be sure, but that's my best hunch currently :)

To be clear, I'm fairly certain this does not affect the release branch at https://ifrc-go-release-v4-3-11.surge.sh/regions/2#operations - can you confirm @nanometrenat ?

**Comment by nanometrenat** [comment]
@batpad confirmed that it does not affect the release branch - thanks for checking :)
xref https://github.com/IFRCGo/go-frontend/issues/1386 as mentioned (whether there's a real link is still being investigated)

**Comment by nanometrenat** [comment]
I am having same symptom on this emergency page in staging:  https://dsgoproxyapp.azurewebsites.net/emergencies/3276#details whereas as at this instant right now the regional page for the Americas (that was getting stuck on loading earlier today) is loading fine.
I tried another emergency (https://dsgoproxyapp.azurewebsites.net/emergencies/3252#details) but am getting a grey screen on that one (i.e. nothing loading at all)

**Comment by batpad** [comment]
Should be fixed with https://github.com/IFRCGo/go-frontend/pull/1495 - thanks @frozenhelium !

**Comment by batpad** [comment]
Am still seeing a grey screen on https://dsgoproxyapp.azurewebsites.net/emergencies/3252#details, which seems to be a separate issue from the loading issue. I will investigate that separately.

**Comment by batpad** [comment]
Just to note, I can reproduce this grey screen for this particular emergency on the Release branch as well: https://ifrc-go-release-v4-3-11.surge.sh/emergencies/3252#details . So this is a release blocker. Seems to be not handling the case of when an Emergency does not have any countries associated with it. Not sure exactly how that case should be handled with regards to breadcrumbs, etc., but going to fix it so that the page does not crash.

@karitotp for the future, this is likely a good emergency to keep a note of to make sure we test, as it does not have any Country associated: https://dsgoproxyapp.azurewebsites.net/emergencies/3252#details

**Comment by batpad** [comment]
Created a separate issue for the emergency grey screen: https://github.com/IFRCGo/go-frontend/issues/1496

**Comment by nanometrenat** [comment]
I was going to close this as I haven't had any emergency or regional pages be stuck on loading since the "stuck on loading" fix went out.

However @batpad @frozenhelium the ticket is still at "To Do" so I'm not sure if there's still therefore further work required before it should be closed?

Thanks a lot
cc @LukeCaley

**Comment by nanometrenat** [comment]
This morning in staging I had https://dsgoproxyapp.azurewebsites.net/emergencies/3983#details stuck on loading - however I am no longer getting stuck on loading for that page.  Noting here for reference (not reopening the ticket if it is a one-off)  Cheers!

**Comment by batpad** [comment]
@frozenhelium I am still a bit worried about the code that handles this :) - I'm moving this to `In Progress` and I'll poke around and see if I can figure out how this might happen. I got on a call with @nanometrenat to look at this and it did not seem like network requests failed, so this seems like it may still be some sort of weird race condition / timing issue that causes the loading spinner to not go away (again, all the data had loaded in the background, but the spinner did not go away). The code for that loading spinner seems like it was always a bit awkward, will hit you up if it helps to talk this through. Thanks much.

**Comment by batpad** [comment]
Also, to be clear to document this: We had a fix for this that went into production that definitely solved this for "almost all the time". There does seem like there "could be" a lingering race condition / timing issue, though this is hard to reproduce. We will take a look, and in the meanwhile if anyone does run into this: Page data seeming to load, but loading spinner not going away, please report here, add to this issue - thanks!

**Comment by nanometrenat** [comment]
@batpad am just using GO in Firefox for the moment, troubleshooting some user issues - and have again got the "stuck on loading" thing - in Production - as per screenshot below:
![image](https://user-images.githubusercontent.com/33625192/90140000-672cec00-dd79-11ea-840a-5e18eae348cb.png)

**Comment by nanometrenat** [comment]
Actually, re the above comment, I spoke too soon - the "loading" eventually went away, but data not available (Firefox only - fine in Chrome):
![image](https://user-images.githubusercontent.com/33625192/90140315-d9053580-dd79-11ea-89a4-a57ef60fd7a8.png)
Presumably not same issue as this ticket after all.

**Comment by nanometrenat** [comment]
Reliably getting "stuck on loading" on this page in staging this morning:  https://dsgoproxyapp.azurewebsites.net/reports/9637

**Comment by nanometrenat** [comment]
Also getting this on Geo Surge URL:  https://ifrc-go-feature-geo-apis.surge.sh/emergencies/3983#details

**Comment by nanometrenat** [comment]
@batpad @frozenhelium I can reliably reproduce this.

Go to https://ifrc-go-feature-geo-apis.surge.sh/emergencies/3983#details in an incognito window (i.e. not logged in).  Page loads fine.  Click Log In button and log in.  Page gets stuck on loading.

**Comment by GregoryHorvath** [comment]
@batpad @frozenhelium @nanometrenat The same thing happens as on the "Steps to reproduce" picture, but with the "Account" page.

Steps:
1. Freshly log in
2. Go to your account dropdown -> Account
3. It now shows the loading forever and the whole page is disabled (can't click anything)
4. Refresh and it goes back to normal

Revisiting the Account page after seems to be working well.

**Comment by nanometrenat** [comment]
Same for emergencies - stuck on loading again (this was previously fixed though right?  but broken as at right now)

- Go to https://dsgoproxyapp.azurewebsites.net/emergencies/3983#additional-information
- Click log in
- Input username and password
- get stuck on emergency page with loading dots

**Comment by nanometrenat** [comment]
Could this possibly (perhaps, hypothetically) be linked to https://github.com/IFRCGo/go-frontend/issues/1386 re the Initialisation, as if the user is NOT logged in when they first hit the site (and get the initialising screen/preloading) then they will have the unauthenticated version of all the data, then when they log in they are missing out on anything that requires authentication e.g. user-specific stuff (as they were not authenticated when the initialisation happened)?  Just a guess but wondering.

@LukeCaley @tovari this is a blocker for site usage and is critical to be fixed and robust before any release to Prod.  Else users get stuck on pages and can't do anything without reloading.

**Comment by GregoryHorvath** [comment]
> @batpad @frozenhelium @nanometrenat The same thing happens as on the "Steps to reproduce" picture, but with the "Account" page.
> 
> Steps:
> 
> 1. Freshly log in
> 2. Go to your account dropdown -> Account
> 3. It now shows the loading forever and the whole page is disabled (can't click anything)
> 4. Refresh and it goes back to normal
> 
> Revisiting the Account page after seems to be working well.

The fix for this is deployed to staging. Looking at the other happenings.

**Comment by LukeCaley** [comment]
@batpad - can you update on the status of this ticket please?  It would be good to know if this is going to be a blocker for the release and to give @karitotp sufficient time and detailed scenarios to test robustly.

**Comment by batpad** [comment]
This has been addressed in #1549 and should fix the issues related to pages getting stuck on loading. Again, thank you @frozenhelium !

We have updated all pages identified as having issues with the new pattern for displaying loading with the new global loading pattern. We have as yet not implemented this on all pages, but we believe this should no longer cause any user facing issues and implementing the new global loading across all pages can be done as a later refactor.

This will be great to test thoroughly. Visit all pages on site, try reloading pages a few times, etc. and check if it is possible to reproduce the state where pages are "stuck" on the loading state.

cc @karitotp
